charts/promtail/templates/_pod.tpl

```diff
@@ -81,6 +81,12 @@ Pod template used in Daemonset and Deployment
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+      {{- if .Values.deployment.enabled }}
```

Comment:
I decided to add this only for deployments as running a sidecar for a DaemonSet seems very wrong 🙃